src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -722,6 +723,58 @@ CreateOffer::applyGuts (ApplyView& view, ApplyView& view_cancel)
 
     if (result == tesSUCCESS)
     {
+        // If a tick size applies, round the offer to the tick size
+        auto const& uPaysIssuerID = saTakerPays.getIssuer ();
+        auto const& uGetsIssuerID = saTakerGets.getIssuer ();
+
+        std::uint8_t uTickSize = Quality::maxTickSize;
+        if (!isXRP (uPaysIssuerID))
+        {
+            auto const sle =
+                view.read(keylet::account(uPaysIssuerID));
+            if (sle && sle->isFieldPresent (sfTickSize))
+                uTickSize = std::min (uTickSize,
+                    (*sle)[sfTickSize]);
+        }
+        if (!isXRP (uGetsIssuerID))
+        {
+            auto const sle =
+                view.read(keylet::account(uGetsIssuerID));
+            if (sle && sle->isFieldPresent (sfTickSize))
+                uTickSize = std::min (uTickSize,
+                    (*sle)[sfTickSize]);
+        }
+        if (uTickSize < Quality::maxTickSize)
+        {
+            auto const rate =
+                Quality{saTakerPays, saTakerGets}.round
```

Comment:
I may have reversed them all the way through. It will work the same that way, but obviously it's setting a trap for maintaining the code. I'll fix it to be consistent.